build.proj

```diff
@@ -21,8 +21,7 @@
 
   <ItemGroup>
     <Project Include="external\dir.proj" />
-    <Project Include="src\ref.builds" />
```

Comment:
It wasn't clear to me that there was an explicit reason to build ref/src outside of dirs.proj other than that it made implementation of vertical builds simpler at the time.  @mellinoe added "SerializeProjects=true" to dirs.proj in https://github.com/dotnet/corefx/commit/a9f6d7d749f9946cc6b10c50b54bcae0e43e859e, and this was restoring the previous method.  If we're going to explicitly call ref.builds/src.builds from build.proj, then we should probably delete dirs.proj